include/CppUTest/CppUTestConfig.h

```diff
@@ -199,6 +199,27 @@
 #define CPPUTEST_CHAR_BIT 8
 #endif
 
+/* Support for "long long" type */
+#ifdef CPPUTEST_USE_LONG_LONG
```

Comment:
I did consider letting it actually copy data around as an opaque memcpy'd type.  But there are technical problems with that (you'd need to add conversion macros, and there's still problems with formatting, especially cross endian).  And even after all of that it wouldn't support large numeric literals, which inevitably someone will want to use in these sorts of tests.

So it makes things very messy for very little gain, since if you're using a compiler that really doesn't support long longs, you're obviously not going to have code that has variables of that type.  Even if it's a cross-platform app where **most** platforms support long long, if there's one that doesn't you'll either have to avoid using it or do some conditional code for that platform, in which case the tests won't need to use long longs on that platform either.

As it stands this seemed like a reasonable compromise: if you define the option then you get support (and things fail to compile if your compiler doesn't support the type, as you'd expect).  If you don't define the option then you don't get support (even if the compiler could support it) but everything else works as expected.

And if you somehow end up defining the option differently between compiling CppUTest and compiling the unit test application, then you get a link error, due to name mangling differences between the symbols.  I'm not sure if there's a good way to avoid this (you probably don't want to avoid it in the case where the lib doesn't support but the app wants it, but the reverse is possibly more reasonable).